agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/statsbeat/NetworkStatsbeat.java

```diff
@@ -241,25 +246,16 @@ private double getRequestDurationAvg() {
 
   /**
    * e.g. endpointUrl 'https://westus-0.in.applicationinsights.azure.com/v2.1/track' host will
-   * return 'westus-0.in.applicationinsights.azure.com'
+   * return 'westus-0'
    */
+  @SuppressWarnings("SystemOut")
   static String getHost(String endpointUrl) {
-    assert (endpointUrl != null && !endpointUrl.isEmpty());
-    int start = endpointUrl.indexOf("://");
-    if (start != -1) {
-      int end = endpointUrl.indexOf("/", start + 3);
-      if (end != -1) {
-        return endpointUrl.substring(start + 3, end);
-      }
-
-      return endpointUrl.substring(start + 3);
-    }
+    Matcher matcher = hostPattern.matcher(endpointUrl);
 
-    int end = endpointUrl.indexOf("/");
-    if (end != -1) {
-      return endpointUrl.substring(0, end);
+    if (matcher.find()) {
+      return matcher.group(1);
     }
 
-    return endpointUrl;
+    return "";
```

Comment:
what do u think of returning the full url as the fallback? that way we could understand why it didn't parse
```suggestion
    return endpointUrl;
```

Reply:
that will end up sending the full endpointUrl as host to statsbeat.

Reply:
isn't this a good thing? then we can see what the problem was?

Reply:
well, if that's the case, telemetry won't get sent at all.. it's from customer's ingestion endpoint.

Reply:
I thought this is just for the attribute on the statsbeat telemetry, probably I'm missing something, we can discuss mon